packages/nimble-components/src/wafer-map/modules/experimental/worker-renderer.ts

```diff
@@ -1,14 +1,65 @@
+import { type Remote, transfer } from 'comlink';
 import type { WaferMap } from '../..';
 import { HoverDieOpacity } from '../../types';
+import { createMatrixRenderer } from '../create-matrix-renderer';
+import type { MatrixRenderer } from '../../../../build/generate-workers/dist/esm/source/matrix-renderer';
 
 /**
  * Responsible for drawing the dies inside the wafer map, adding dieText and scaling the canvas
  */
 export class WorkerRenderer {
+    private worker!: Remote<MatrixRenderer>;
+
+    private isWorkerAlive = false;
+
     public constructor(private readonly wafermap: WaferMap) {}
 
-    public drawWafer(): void {
-        // rendering will be implemented in a future PR
+    public async setupAndDrawWafer(): Promise<void> {
+        if (this.wafermap.diesTable === undefined) {
+            return;
+        }
+        await this.setupWorker();
+        const columnIndexes = Int32Array.from(
+            this.wafermap.diesTable.getChild('colIndex')!.toArray()
+        );
+        const rowIndexes = Int32Array.from(
+            this.wafermap.diesTable.getChild('rowIndex')!.toArray()
+        );
```

Comment:
If I understand correctly this results in iterating across the data (potentially pretty slowly?) and making a copy into a new array. I'm a bit surprised by making a copy, is that intentional long-term?

Reply:
I wasn't remembering correctly, I modified the code as you suggested https://github.com/ni/nimble/pull/1929/commits/5c74f692678137e2ab454535360e750a544c259f